pkg/providers/amifamily/windows.go

```diff
@@ -0,0 +1,84 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package amifamily
+
+import (
+	"fmt"
+
+	"github.com/aws/karpenter-core/pkg/scheduling"
+
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/karpenter/pkg/providers/amifamily/bootstrap"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+)
+
+type Windows struct {
+	DefaultFamily
+	*Options
+	Version string
+	Variant string
+}
+
+func (w Windows) DefaultAMIs(version string) []DefaultAMIOutput {
+	return []DefaultAMIOutput{
+		{
+			Name:  fmt.Sprintf("windows-server-%s-%s-%s", w.Version, w.Variant, version),
+			Query: fmt.Sprintf("/aws/service/ami-windows-latest/Windows_Server-%s-English-%s-EKS_Optimized-%s/image_id", w.Version, w.Variant, version),
+			Requirements: scheduling.NewRequirements(
```

Comment:
There should also be a `windows-build` requirement here since if the build number doesn't match for any deployment that is going to be joining this node, the AMI will be incompatible.

Reply:
Used as const to create SSM query.